snooty/taxonomy.py

```diff
@@ -42,11 +41,14 @@ def get_taxonomy(cls) -> "TaxonomySpec":
         return cls.TAXONOMY_SPEC
 
     @classmethod
-    def validate_key_value_pairs(cls, facet_str_pairs: List[Tuple[str, str]]) -> None:
+    def validate_key_value_pairs(
+        cls, facet_str_pairs: Sequence[Tuple[str, str]]
+    ) -> None:
         taxonomy_ref = asdict(cls.get_taxonomy())
+        facet_pairs = list(facet_str_pairs[:])
```

Comment:
Oh, if you need a list, then `list()` already does a copy, so this should just be `list(facet_str_pairs)`